src/java/htsjdk/samtools/fastq/FastqRecord.java

```diff
@@ -23,62 +23,144 @@
  */
 package htsjdk.samtools.fastq;
 
+import htsjdk.samtools.Read;
+import htsjdk.samtools.SAMUtils;
+import htsjdk.samtools.util.StringUtil;
+
 import java.io.Serializable;
 
 /**
- * Represents a fastq record, fairly literally, i.e. without any conversion.
+ * Simple representation of a FASTQ record, without any conversion
  */
-public class FastqRecord implements Serializable {
+public class FastqRecord implements Read, Serializable {
     private static final long serialVersionUID = 1L;
-    private final String seqHeaderPrefix;
-    private final String seqLine;
-    private final String qualHeaderPrefix;
-    private final String qualLine;
-
-    public FastqRecord(final String seqHeaderPrefix, final String seqLine, final String qualHeaderPrefix, final String qualLine) {
-        if (seqHeaderPrefix != null && !seqHeaderPrefix.isEmpty()) this.seqHeaderPrefix = seqHeaderPrefix;
-        else this.seqHeaderPrefix = null;
-        if (qualHeaderPrefix != null && !qualHeaderPrefix.isEmpty()) this.qualHeaderPrefix = qualHeaderPrefix;
-        else this.qualHeaderPrefix = null;
-        this.seqLine = seqLine ;
-        this.qualLine = qualLine ;
+    private final String readName;
+    private final String readString;
+    private final String qualityHeader;
+    private final String qualityString;
+
+    /**
+     * Default constructor
+     *
+     * @param readName      the read name
+     * @param readStrig     the sequence string
+     * @param qualityHeader the quality header
+     * @param qualityString the quality string
+     */
+    public FastqRecord(final String readName, final String readStrig, final String qualityHeader, final String qualityString) {
+        if (readName != null && !readName.isEmpty()) this.readName = readName;
+        else this.readName = null;
+        if (qualityHeader != null && !qualityHeader.isEmpty()) this.qualityHeader = qualityHeader;
+        else this.qualityHeader = null;
+        this.readString = readStrig;
+        this.qualityString = qualityString;
     }
-    
-    /** copy constructor */
+
+    /**
+     * Constructor for byte[] arrays
+     *
+     * @param readName      the read name
+     * @param readBases     the read sequence as ASCII bytes ACGTN=.
+     * @param qualityHeader the quality header
+     * @param readQualities the base qualities as binary PHRED scores (not ASCII)
+     */
+    public FastqRecord(final String readName, final byte[] readBases, final String qualityHeader, final byte[] readQualities) {
+        this(readName, StringUtil.bytesToString(readBases), qualityHeader, SAMUtils.phredToFastq(readQualities));
+    }
+
+    /**
+     * Copy constructor from a read
+     *
+     * @param read the read to convert to a FastqRecord
+     */
+    public FastqRecord(final Read read) {
+        this(read.getReadName(), read.getReadString(), (read instanceof FastqRecord) ? ((FastqRecord) read).qualityHeader : null, read.getBaseQualityString());
```

Comment:
It is true that it is the same value, but the problem is that the qualityHeader could contain some information that it is not in the readHeader. For instance, you can have a readHeader only with the read name and the qualityHeader with some other information like read group and other stuff.

I'm actually don't know if anyone is using the qualityHeader, because in my experience this one does not have any value...